src/html/html-document.ts

```diff
@@ -102,21 +103,42 @@ export class ParsedHtmlDocument extends ParsedDocument<ASTNode, HtmlVisitor> {
     };
   }
 
-  stringify() {
-    // TODO(issues/315): serialize inline documents correctly.
-    let result = parse5.serialize(this.ast);
+  stringify(options?: StringifyOptions) {
+    options = options || {};
+    // We want to make of copy of `this` and all of the inline documents such
```

Comment:
Yeah, this is the craziest trick I've pulled in a little while. Expanded this comment out a bunch.